go/libraries/doltcore/servercfg/yaml_config.go

```diff
@@ -249,7 +247,149 @@ func clusterConfigAsYAMLConfig(config ClusterConfig) *ClusterYAMLConfig {
+// If an empty field has a default value, the default will be used.
+// If an empty field has no default value, a commented-out placeholder will be used.
+func (cfg YAMLConfig) VerboseString() string {
+	withDefaults := cfg.withDefaultsFilledIn()
+
+	formatted := formattedYAMLMarshal(removeOmitemptyTags(withDefaults))
+	formatted = commentNullYAMLValues(formatted)
+
+	return formatted
+}
+
+// withDefaultsFilledIn returns the config with default values in place of nil values.
+func (cfg YAMLConfig) withDefaultsFilledIn() YAMLConfig {
+	defaults := defaultServerConfigYAML()
+	withDefaults := cfg
+
+	if withDefaults.LogLevelStr == nil {
+		withDefaults.LogLevelStr = defaults.LogLevelStr
+	}
+	if withDefaults.MaxQueryLenInLogs == nil {
+		withDefaults.MaxQueryLenInLogs = defaults.MaxQueryLenInLogs
+	}
+	if withDefaults.EncodeLoggedQuery == nil {
+		withDefaults.EncodeLoggedQuery = defaults.EncodeLoggedQuery
+	}
+
+	if withDefaults.BehaviorConfig.ReadOnly == nil {
+		withDefaults.BehaviorConfig.ReadOnly = defaults.BehaviorConfig.ReadOnly
+	}
+	if withDefaults.BehaviorConfig.AutoCommit == nil {
+		withDefaults.BehaviorConfig.AutoCommit = defaults.BehaviorConfig.AutoCommit
+	}
+	if withDefaults.BehaviorConfig.DoltTransactionCommit == nil {
+		withDefaults.BehaviorConfig.DoltTransactionCommit = defaults.BehaviorConfig.DoltTransactionCommit
+	}
+
+	if withDefaults.UserConfig.Name == nil {
+		withDefaults.UserConfig.Name = defaults.UserConfig.Name
+	}
+	if withDefaults.UserConfig.Password == nil {
+		withDefaults.UserConfig.Password = defaults.UserConfig.Password
+	}
+
+	if withDefaults.ListenerConfig.HostStr == nil {
+		withDefaults.ListenerConfig.HostStr = defaults.ListenerConfig.HostStr
+	}
+	if withDefaults.ListenerConfig.PortNumber == nil {
+		withDefaults.ListenerConfig.PortNumber = defaults.ListenerConfig.PortNumber
+	}
+	if withDefaults.ListenerConfig.MaxConnections == nil {
+		withDefaults.ListenerConfig.MaxConnections = defaults.ListenerConfig.MaxConnections
+	}
+	if withDefaults.ListenerConfig.ReadTimeoutMillis == nil {
+		withDefaults.ListenerConfig.ReadTimeoutMillis = defaults.ListenerConfig.ReadTimeoutMillis
+	}
+	if withDefaults.ListenerConfig.WriteTimeoutMillis == nil {
+		withDefaults.ListenerConfig.WriteTimeoutMillis = defaults.ListenerConfig.WriteTimeoutMillis
+	}
+	if withDefaults.ListenerConfig.AllowCleartextPasswords == nil {
+		withDefaults.ListenerConfig.AllowCleartextPasswords = defaults.ListenerConfig.AllowCleartextPasswords
+	}
+
+	if withDefaults.DataDirStr == nil {
+		withDefaults.DataDirStr = defaults.DataDirStr
+	}
+	if withDefaults.CfgDirStr == nil {
+		withDefaults.CfgDirStr = defaults.CfgDirStr
+	}
+	if withDefaults.PrivilegeFile == nil {
+		withDefaults.PrivilegeFile = defaults.PrivilegeFile
+	}
+	if withDefaults.BranchControlFile == nil {
+		withDefaults.BranchControlFile = defaults.BranchControlFile
+	}
+
+	return withDefaults
+}
+
+// Assumes 'in' has no circular references.
+func removeOmitemptyTags(in any) any {
```

Comment:
I think we have the same comment here: we'd rather just avoid the use of reflection altogether and do this manually. Just list the fields that have `omitempty` manually rather than examining them via reflection.

Reply:
See also my other comment in the test file. `omitempty` isn't the best determinant of whether to comment out a line of config in the file produced.